pkl-core/src/main/java/org/pkl/core/externalreader/ResourceResolver.java

```diff
@@ -13,24 +13,20 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.pkl.core.resource;
+package org.pkl.core.externalreader;
 
 import java.io.IOException;
 import java.net.URI;
 import java.util.List;
 import java.util.Optional;
 import org.pkl.core.SecurityManager;
 import org.pkl.core.SecurityManagerException;
+import org.pkl.core.messaging.MessageTransport;
 import org.pkl.core.module.PathElement;
 
-public interface ExternalResourceResolver {
-
-  interface Spec {
-    boolean hasHierarchicalUris();
-
-    boolean isGlobbable();
-
-    String scheme();
+public interface ResourceResolver {
+  static ResourceResolver of(MessageTransport transport, long evaluatorId) {
+    return new ResourceResolverImpl(transport, evaluatorId);
```

Comment:
Same comment; let's keep this class as-is, and have `ResourceReaderSpec` implement this interface